pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/formats/protobufnative/PulsarProtobufNativeFormatFactory.java

```diff
@@ -105,17 +107,35 @@ public ChangelogMode getChangelogMode() {
     }
 
     private String extractTopicName(DynamicTableFactory.Context context) {
-
         validateTopic(context.getCatalogTable().getOptions());
-        String topic = context.getCatalogTable().getOptions().get(TOPIC.key());
-        // Maybe catalog mode
-        if (StringUtils.isNullOrWhitespaceOnly(topic)) {
-            final ObjectIdentifier table = context.getObjectIdentifier();
-            topic = TopicName.get(table.getDatabaseName() + "/" + table.getObjectName()).toString();
+
+        String topic;
+        if (isGenericTable(context)) {
+            topic = context.getCatalogTable().getOptions().get(TOPIC.key());
```

Comment:
If it is a generic table

1. use getCatalogTable() refers to the orignal native pulsar topic (not in __flink_catalog tenant)
2. objectNameToTopicName() refers to the mapped topic (under __flink_catalog tenant)
I think here what we want is the topic carrying flink schema ? So maybe we should use 2nd approach ?